detector/labeler.py

```diff
@@ -35,20 +37,22 @@ class EnronLabeler:
     def __init__(
         self, 
         data: pd.DataFrame = None,
+        needs_preprocessing: bool = False,
```

Comment:
@FalgunMalhotra added this flag that decides whether to run preprocessing in labeler or not. Default to False